### DISCUSSION TITLE
Multiple sensors for the same light?

### DISCUSSION THREAD

**Discussion by mariomaz87** [Jason-Morcos/homebridge-magic-occupancy#28] [Q&A]
Hi, I would like to use multiple sensors to keep on a light or turn it off when no movement is detected (I have a very large patio with many sensors and one light switch, I want the light to stay on when there is movement on any of the sensors). I tried to replicate the same routines I have with other rooms but it’s not working as expected (the light starts a cycle of on\off). I am missing something? Thanks!

**Comment by Jason-Morcos** [comment]
This should totally be doable with this plug-in! I'm doing it myself. I set up one switch per motion sensor, all part of the same occupancy sensor (just comma separate the names for the switches themselves in the settings UI). Do you have it set up differently than that?

**Reply by mariomaz87** [reply]
Thanks, probably this is the issue. I set up only one switch, and all the motion sensors are turning on and off the same switch with the same automations. 
If I understand correctly your suggestion, in the case of two motion sensors for example you have:
Motion Sensor A detects motion > turn on Magic Switch A
Motion Sensor B detects motion > turn on Magic Switch B
Motion Sensor A does not detect motion > turn off Magic Switch A
Motion Sensor B does not detect motion > turn off Magic Switch B

and then
Magic Sensor A turns on > Occupancy Room on
Magic Sensor B turns on > Occupancy Room on
Magic Sensor A turns off > Occupancy Room off
Magic Sensor B turns off > Occupancy Room off

Is it correct?
Thanks again!

**Reply by Jason-Morcos** [reply]
Sorta, I think we may be saying the same thing but getting a little confused on the semantics!

Starting from the plugin config, you'd need to create an occupancy sensor with two stateful switches.
<img width="772" alt="Screen Shot 2022-04-27 at 09 39 18" src="https://user-images.githubusercontent.com/10710367/165573074-6ded761b-fb0a-4181-b19f-6ff986f7db59.png">

Then, create automations in HomeKit:
- Motion Sensor A detects motion > turn on MAGIC1
- Motion Sensor B detects motion > turn on MAGIC2
- Motion Sensor A does not detect motion > turn off MAGIC1
- Motion Sensor B does not detect motion > turn off MAGIC2
- Example Occupancy sensor detects occupancy > turn on the light
- Example Occupancy sensor does not detect occupancy > turn off the light


Doing that should mean that as long as either of the motion sensors detect motion, the light will stay on. You can adjust how long the light should stay on after motion stops by changing the Stay Occupied Delay

**Reply by mariomaz87** [reply]
Thanks, this is working!
